arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -921,6 +921,35 @@ uint64_t MaintenanceFeature::shardVersion(std::string const& shname) const {
 
 uint64_t MaintenanceFeature::incShardVersion(std::string const& shname) {
   MUTEX_LOCKER(guard, _versionLock);
+
+  uint64_t v = 0;
+  using namespace std::chrono;
+  using clock = steady_clock;
+  auto timeout = duration<double>(600.0);
+  auto stoppage = clock::now();
+  auto snooze = milliseconds(100);
+
+  // Acquire current version from agency and wait for it to have been dealt
+  // with in local current cache. Any future current version will do, as
+  // the version is incremented by the leader ahead of getting here on the
+  // follower.
+  while (!server().isStopping() && clock::now() < stoppage) {
+    cluster::fetchCurrentVersion(0.1 * timeout)
+      .thenValue(
+        [&v] (auto&& res) { v = res.get(); })
```

Comment:
when we get the value here, this does not abort the loop??

Reply:
We also do not wait for the future here at all, so we will re-fire the request every 100ms regardless. You can use the `wait_for` method on the future instead of a raw sleep?

Reply:
I think this is a bug. `fetchCurrentVersion` returns a future, we need to explicitly wait for its completion here and then handle the two cases that it has worked (within the timeout) or not.